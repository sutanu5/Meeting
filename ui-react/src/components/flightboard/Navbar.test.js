import Navbar from './Navbar';

describe('Flightboard Navbar', () => {
  let mockFilter, wrapper;

  beforeEach(() => {
    mockFilter = jest.fn();
    wrapper = shallow(<Navbar filter={mockFilter} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders navbar-title with text', () => {
    const title = wrapper.find('#fb__navbar-title').text();

    expect(title).not.toBeEmpty();
  });

  it('renders RoomFilterContainer component correctly', () => {
    const filter = wrapper.find('RoomFilterContainer').exists();

    expect(filter).toBeTruthy();
  });

  it('renders Clock component correctly', () => {
    const clock = wrapper.find('Clock').exists();

    expect(clock).toBeTruthy();
  });

  it('operates filter callback function correctly', () => {
    wrapper.find('RoomFilterContainer').prop('filter')('foo');

    expect(mockFilter).toHaveBeenCalledWith('foo');
  });
});
