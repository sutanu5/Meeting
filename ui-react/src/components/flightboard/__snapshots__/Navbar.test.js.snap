// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard Navbar renders correctly 1`] = `
<div
  id="title-bar-wrap"
>
  <div
    className="title-bar"
  >
    <div
      className="title-bar-left"
    >
      <ul
        className="horizontal menu fb__nav-menu"
      >
        <li>
          <img
            alt="Logo"
            src="img/logo.png"
          />
        </li>
        <li>
          <span
            id="fb__navbar-title"
          >
            Conference Room Availability
          </span>
          <br />
          <ul
            className="dropdown menu fb__dropdown"
            data-dropdown-menu={true}
            id="roomlist-filter"
          >
            <RoomFilterContainer
              filter={[Function]}
            />
          </ul>
        </li>
      </ul>
    </div>
    <div
      className="title-bar-right"
    >
      <ul
        className="horizontal menu fb__nav-menu-right"
      >
        <li
          id="the-clock"
        >
          <Clock />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
